### ISSUE TITLE
openmp fails to build for clangarm64

### ISSUE THREAD

**Issue by lazka** [msys2/MINGW-packages#13818]
Future reference so we know what the status is:

* https://packages.msys2.org/base/mingw-w64-openmp
* openmp doesn't build for aarch64 right now (llvm 15.0.3)
* In theory there is aarch64 support upstream, but there are likely MSVC specifics left, or just some preprocessor guards have to be adjusted for mingw, or something completely different: https://github.com/llvm/llvm-project/commit/4fb0aaf03381473ec8af727edb4b5d59b64b0d6

Not having openmp isn't that much of a blocker for arm64, but any help is welcome :)

Current build error:

```
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_10
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_20
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_1_cas
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_1_cas_cpt
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_2_cas
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_2_cas_cpt
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_4_cas
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_4_cas_cpt
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_8_cas
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_bool_8_cas_cpt
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_cmplx10_add
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_cmplx10_div
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_cmplx10_mul
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_cmplx10_sub
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_float10_add
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_float10_div
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_float10_max
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_float10_max_cpt
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_float10_min
ld.lld: error: <root>: undefined symbol: __kmpc_atomic_float10_min_cpt
ld.lld: error: too many errors emitted, stopping now (use /errorlimit:0 to see all errors)
```

**Comment by Biswa96** [comment]
Does it require or compile any assembly code while building in aarch64?

**Comment by mati865** [comment]
Dunno but the referenced commit is "the first in the series..." but there were no further patches from that account....
Missing symbols look like i686/amd64 specific: https://cs.github.com/llvm/llvm-project/blob/bcedeefa40a4759c40379d957f38da2b10a14f63/openmp/runtime/src/kmp_atomic.cpp#L3607

**Comment by lazka** [comment]
Oh, I found upstream issue: https://github.com/llvm/llvm-project/issues/56349

**Comment by lazka** [comment]
Fixed via #13933 and 93acfc650c04fb4
